### PR TITLE
minor fixes and new range feature

### PR DESCRIPTION
**Pull request by cltrudeau**
Hi Jim,

I've made some small fixes and added one new feature for ranges. 

* Changed setup.py to include pyperclip and made the dependency versions explicit
* Changed the -c and -v options to use @clip's "is_flag", the use of the boolean meant you had to pass in a boolean paramater, now the presence of the flag is good enough
* Added "$" as a valid range specifier
* Added "$-3" negative indexing (last line and three proceeding it)
* Change examples/testfile.py to contain code and add import_function to the simple.mdt file (I missed that import_function existed when I wrote my last article because I just skimmed the example and stopped)

Your code, your lib, so feel free to reject anything you like.

### CODE REVIEW

File: markplates/__main__.py [hunk 1]
```diff
@@ -224,13 +238,12 @@ def process_template(template):
 
 @click.command(context_settings=dict(help_option_names=["-h", "--help"]))
 @click.option(
-    "-v", "--verbose", type=bool, default=False, help="Verbose debugging info"
+    "-v", "--verbose", is_flag=True, help="Verbose debugging info"
```
Comment (jima80525) [inline markplates/__main__.py:241]: nice catch.  I'm surprised I didn't notice this. thanks for fixing!

File: setup.py [hunk 2]
```diff
@@ -28,6 +28,10 @@
     ],
     packages=[NAME],
     include_package_data=False,
-    install_requires=["Click", "Jinja2"],
+    install_requires=[
+        "Click==7.1.2", 
+        "Jinja2==2.11.2",
+        "pyperclip==1.8.0",
+    ],
```
Comment (jima80525) [inline setup.py:35]: This is probably what I was missing to get pyperclip running in tox.  I only added the -c option to work around this limitation (I was rushing to write my article at the time. :) )   Do you have an opinion on removing the -c and just making it always copy to the clipboard?

File: README.md [hunk 3]
```diff
@@ -102,7 +107,7 @@ Examples:
 
 ### `import_function()`
 
-The `import_function` function will search the source file and include only the specified function. If there are multiple functions with the same name in the source_file, only the first will be included (and you shouldn't have multiple functions with the same name anyway!).
+The `inport_function` function will search the source file and include only the specified function. If there are multiple functions with the same name in the source_file, only the first will be included (and you shouldn't have multiple functions with the same name anyway!).
```
[the remark this answers is not in the document]
Comment (cltrudeau) [reply README.md:110]: I think I may have accidentally, out of habit, edited the README.md file then possibly overwritten my own changes generating from the template at some point. I'll read the whole thing top to bottom again and then commit

File: setup.py (hunk 2 above)
Comment (cltrudeau) [reply setup.py:35]: Funny you say that, I kind of wondered why it needed an option. Was going to write a bash wrapper to put it on all the time for myself anyhow :)

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (cltrudeau) [reply README.md:110]: Nope, I'm not sure what happened here. The changes I thought I put in are on the other branch in the other PR. Sorry if I mucked your correction. I don't think anything else got touched.

[the remark this answers is not in the document]
Comment (jima80525) [reply README.md]: I just merged the other PR, which was a superset of these, I believe.  OK to just close this one?

**Comment by jima80525** [comment]
Hey!  I Forgot this was sitting around waiting for me!  Sorry!  I'm in the process of making a new article and have some updates to add myself and just noticed.  Got these merged and I'll add my changes on top.  New changes coming in are fixing some error reporting (esp when you specify a range that goes beyond the end of the file) and adding a template option to add a `# ...` line if there's a gap in the range to indicate you skipped some of the file.

**Pull request merged**